census/controllers/census.js

```diff
@@ -219,6 +221,39 @@ var submit = function(req, res) {
   });
 };
 
+let _getDiscussionURL = function(req, dataset, place) {
+  /*
+    If `submission_discussion_url` is defined in settings and it is in the
+    format: https://discuss.okfn.org/c/<topic>/<subtopic>, return a new topic
+    url with a prepopulated topic for place and dataset. Otherwise, return the
+    original `submission_discussion_url` without modification. If
+    `submission_discussion_url` is undefined return an empty string.
+  */
+  let submissionDiscussionURL =
+    _.get(req.params.site.settings, 'submission_discussion_url', '');
+  let parsedURL = url.parse(submissionDiscussionURL);
+  // URL is a discourse link
+  if (parsedURL.hostname === 'discuss.okfn.org') {
+    let splitPathName = _.trimLeft(parsedURL.pathname, '/').split('/');
+    // URL is a category link
+    if (splitPathName[0] === 'c') {
+      // Create a new topic link
+      let newTopicURL = url.parse('');
+      newTopicURL.protocol = parsedURL.protocol;
+      newTopicURL.host = parsedURL.host;
+      newTopicURL.pathname = 'new-topic';
+      newTopicURL.search = querystring.stringify({
+        title: util.format('Entry for %s / %s', dataset, place),
+        body: util.format('This is a discussion about the submission for [%s / %s](%s).',
```

Comment:
You could also use template strings, which is more javascriptish than this. But it's no issue, more of a convention kind of thing, depends on what's the consensus throughout the code base.